src/app/course/practice-problem/practice-problem.component.ts

```diff
@@ -45,11 +47,16 @@ export class PracticeProblemComponent implements OnInit, OnDestroy {
         private courseService: CourseService,
         private categoryService: CategoryService,
         private userStatsService: UserStatsService,
+        private questionService: QuestionService,
     ) {
     }
 
     ngOnInit(): void {
-        this.difficulty = this.route.snapshot.queryParamMap.get('difficulty')
+
+        this.questionService.getQuestion(this.currentQuestionId).subscribe(result => {
```

Comment:
I don't understand why this line is here! Can you please explain what is its purpose?

Reply:
This is from when I was trying to the current difficulty to show in the dropdown. I got rid of it.